src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -959,93 +987,82 @@ private AnonymousContentAccessCertificate createAnonContentAccessCertificate(
+
+        org.candlepin.model.Content converted = new org.candlepin.model.Content()
+            .setId(cinfo.getId())
+            .setName(cinfo.getName())
+            .setType(cinfo.getType())
+            .setLabel(cinfo.getLabel())
+            .setVendor(cinfo.getVendor())
+            .setGpgUrl(cinfo.getGpgUrl())
+            .setMetadataExpiration(cinfo.getMetadataExpiration())
+            .setRequiredTags(cinfo.getRequiredTags())
+            .setArches(cinfo.getArches())
+            .setContentUrl(cinfo.getContentUrl());
+
+        ProductContent productContent = new ProductContent();
+        productContent.setContent(converted);
+        productContent.setEnabled(pcinfo.isEnabled());
+        return productContent;
     }
 
     /**
-     * Converts {@link ContentInfo} to {@link org.candlepin.model.Content}.
+     * Translates a collection of {@link ProductContent} instances into a list of {@link Content} where the
+     * only populated field is the path (contentUrl), for the purpose of generating the X509 part of the
+     * cert ('Authorized content Urls' is the only content-related information in that part of the cert).
      *
-     * @param contentInfo
-     *  the content to convert
+     * @param productContentSet A collection of ProductContent instances
      *
-     * @return the converted {@link org.candlepin.model.Content} objects
+     * @return A list of Content dtos with only the path populated
      */
-    private List<org.candlepin.model.Content> convertContentInfoToContent(
-        Collection<ContentInfo> contentInfo) {
-        return contentInfo.stream()
-            .filter(Objects::nonNull)
-            .map(content -> {
-                org.candlepin.model.Content converted = new org.candlepin.model.Content();
-                converted.setId(content.getId());
-                converted.setName(content.getName());
-                converted.setType(content.getType());
-                converted.setLabel(content.getLabel());
-                converted.setVendor(content.getVendor());
-                converted.setGpgUrl(content.getGpgUrl());
-                converted.setMetadataExpiration(content.getMetadataExpiration());
-                converted.setRequiredTags(content.getRequiredTags());
-                converted.setArches(content.getArches());
-                converted.setContentUrl(content.getContentUrl());
-                converted.setReleaseVersion(content.getReleaseVersion());
-
-                return converted;
+    public List<Content> convertProductContentToContentDto(Collection<ProductContent> productContentSet) {
+        return productContentSet.stream()
+            .map(productContent -> {
+                Content contentContainer = new Content();
+                contentContainer.setPath(productContent.getContent().getContentUrl());
+                return contentContainer;
             })
-            .toList();
+            .collect(Collectors.toList());
```

Comment:
nit: `.toList()` does this without needing to explicitly import/use `Collectors`

Reply:
Updated